src/app/features/habit/habitSlice.ts

```diff
@@ -0,0 +1,75 @@
+import { createSlice } from "@reduxjs/toolkit";
+import type { PayloadAction } from "@reduxjs/toolkit";
+
+export const habit = {
+  id: null,
+  goal: "",
+  schema: "",
+  startDate: null,
+  rewards: [],
+  habitStrength: {
+    strength: 0,
+    history: [],
+  },
+  data: [],
+};
+
+export interface HabitData {
+  id: number;
+  done?: boolean;
+  reward: boolean;
+}
+
+export interface HabitSlice {
+  id: number | null;
+  goal: string;
+  schema: string;
+  startDate: number | null;
+  rewards: string[];
+  habitStrength: {
+    strength: number;
+    history: { strength: number; date: Date }[];
+  };
+  data: HabitData[];
+}
+
+const savedHabit = localStorage.getItem("habit");
+
+const initialState: HabitSlice = savedHabit ? JSON.parse(savedHabit) : habit;
+const updateSavedHabit = (updatedHabit: HabitSlice) => {
+  localStorage.setItem("habit", JSON.stringify(updatedHabit));
+};
+
+export const habitSlice = createSlice({
+  name: "habit",
+  initialState,
+  reducers: {
+    toggleDone: (state, action: PayloadAction<number>) => {
+      const dayToToggle = state.data.find((el) => el.id === action.payload);
+
+      if (dayToToggle!.done) {
+        dayToToggle!.done = false;
+      } else {
+        dayToToggle!.done = true;
+      }
+      updateSavedHabit(state);
```

Comment:
This thing looks like an effect that will apply for all actions. Maybe there is a better technique to call it whenever this slice changes?